request_handlers.cc

```diff
@@ -212,56 +212,59 @@ RequestHandler::Status ReverseProxyRequestHandler::handleRequest(const Request&
   boost::system::error_code ec;
 
   // Attempt to query the host using the HTTP protocol
-  std::string port = "http"; 
+  std::string port = "http";
   if (reverse_proxy_port != "") {
     port = reverse_proxy_port;
   }
-  std::string modified_proxy_host = 
-  ReverseProxyRequestHandler::sanitizeProxyHost(reverse_proxy_host); 
-  
+  std::string modified_proxy_host =
+  ReverseProxyRequestHandler::sanitizeProxyHost(reverse_proxy_host);
+
+  // Avoid "host not found" error
+  // See: https://stackoverflow.com/questions/12542460/boost-asio-host-not-found-authorative
+  // And: https://github.com/UCLA-CS130/Team05/blob/544269bb60098629000bac8a1278a0c26c0ed7da/reverse_proxy_handler.cc
   tcp::resolver::query query(modified_proxy_host, port);
   tcp::resolver::iterator endpoint_iterator = resolver.resolve(query);
-  
+
   // Try each endpoint until we succesfully establish a connection
   tcp::socket socket(io_service);
   boost::asio::connect(socket, endpoint_iterator, ec);
   if (ec) {
-    // TODO: Return 5xx error instead of 404 
-    std::cout << "Error establishing connection with reverse proxy host: " 
+    // TODO: Return 5xx error instead of 404
```

Comment:
This is an enhancement for later, as `Not Found` is not the same thing as an `Internal Server Error`.